### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Leah2356**

### CODE REVIEW

File: src/main/java/json/RijksService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package json;
+import io.reactivex.rxjava3.core.Single;
+import retrofit2.Retrofit;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public interface RijksService {
+    @GET("/api/en/collection?key={api-key}&p={page}")
+    Single<ArtObjects> pageSearch(@Query("page")int page);
```
Comment (andrewoid) [inline src/main/java/json/RijksService.java]: Your API key and p are query parameters. You use @Query with those. Not "page"

File: src/test/java/feiner/rijks/RijksServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import json.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class RijksServiceTest {
+@Test
+    void pageSearch() {
```
Comment (andrewoid) [inline src/test/java/feiner/rijks/RijksServiceTest.java:13]: You aren't actually passing an apikey in your tests

File: src/main/java/json/ArtObjects.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package json;
+
+public class ArtObjects {
```
Comment (andrewoid) [inline src/main/java/json/ArtObjects.java]: Why is this plural?

File: src/test/java/feiner/rijks/RijksServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        assertNotNull(artObjectCollection.artObjects);
+        ArtObjects artObjects = artObjectCollection.artObjects[0];
+        assertNotNull(artObjects);
+        assertNotNull(artObjects.title);
+        assertNotNull(artObjects.longTitle);
+        assertNotNull(artObjects.principalOrFirstMaker);
+        assertNotNull(artObjects.webImage);
+        WebImage webImage = artObjects.webImage;
+        assertNotNull(webImage.url);
+        assertNotEquals(-1, webImage.offsetPercentageX);
```
Comment (andrewoid) [inline src/test/java/feiner/rijks/RijksServiceTest.java]: 0 not -1

File: src/test/java/feiner/rijks/RijksServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        assertNotNull(webImage.guid);
+        assertNotEquals(-1, webImage.width);
+        assertNotEquals(-1, webImage.height);
+        assertNotEquals(-1, webImage.offsetPercentageY);
+    }
+
+    @Test
+    void querySearch() {
+    // given
```
Comment (andrewoid) [inline src/test/java/feiner/rijks/RijksServiceTest.java]: Indentation is wrong

**Comment by Leah2356** [comment]
When I put in 0, the tests failed. Also, ArtObjects is plural because
ArtObject is already in use in the test: ArtObjects artObject =
artObjectCollection.artObjects[0];

File: src/main/java/json/ArtObjects.java (hunk 3 above)
Comment (andrewoid) [reply src/main/java/json/ArtObjects.java]: This is a singular piece of art, it should be singular and it won't conflict with anything

File: src/test/java/feiner/rijks/RijksServiceTest.java (hunk 4 above)
Comment (andrewoid) [reply src/test/java/feiner/rijks/RijksServiceTest.java]: Put your replies in the pull request, do not reply to the email

Comment (Leah2356) [reply src/test/java/feiner/rijks/RijksServiceTest.java]: When I put in 0, the tests failed.

Comment (andrewoid) [reply src/test/java/feiner/rijks/RijksServiceTest.java]: You don't need the offset percentages but you should have width and height

File: src/main/java/json/WebImage.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package json;
+
+public class WebImage {
+    public String guid;
+    public int offsetPercentageX;
+    public int offsetPercentageY;
```
Comment (andrewoid) [inline src/main/java/json/WebImage.java]: I meant you can remove it completely

File: src/test/java/feiner/rijks/RijksServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        assertNotNull(artObjects);
+        assertNotNull(artObjects.title);
+        assertNotNull(artObjects.longTitle);
+        assertNotNull(artObjects.principalOrFirstMaker);
+        assertNotNull(artObjects.webImage);
+        WebImage webImage = artObjects.webImage;
+        assertNotNull(webImage.url);
+        assertNotNull(webImage.guid);
+        assertNotEquals(-1, webImage.width);
+        assertNotEquals(-1, webImage.height);
```
Comment (andrewoid) [inline src/test/java/feiner/rijks/RijksServiceTest.java]: This will never be false. Make sure they are not 0

**Review by andrewoid — APPROVED** [review]

**Pull request merged**